### PR TITLE
fix(site): Hide agent lifecycle unless opted in via `delay_login_until_ready`

### PR DESCRIPTION
**Pull request by mafredri**
Refs: #5835, #5749

### CODE REVIEW

File: site/src/components/Resources/AgentStatus.tsx [hunk 1]
```diff
@@ -135,22 +135,32 @@ const StartErrorLifecycle: React.FC<{
...
-      <Cond>
-        <StartingLifecycle />
-      </Cond>
-    </ChooseOne>
-  )
+  // NOTE(mafredri): Keep this behind feature flag for the time-being,
+  // if delay_login_until_ready is true, the user has updated to
+  // terraform-provider-coder v0.6.7 and opted in to the functionality.
+  //
+  // Remove check once documentation is in place and we we do a breaking
```
Comment (mafredri) [inline site/src/components/Resources/AgentStatus.tsx:142]:
```suggestion
  // Remove check once documentation is in place and we do a breaking
```

**Review by BrunoQuaresma — APPROVED** [review]
Looks good to me. Just check if the screenshots in Chromatic checks are the expected results

**Comment by mafredri** [comment]
> Looks good to me. Just check if the screenshots in Chromatic checks are the expected results

@BrunoQuaresma the only changes I see is build progress bar. There are no changes to the agent status and this is what I'd expect since the mocked values have `delay_login_until_ready: true`.

**Comment by BrunoQuaresma** [comment]
@mafredri I have to fix this on storybook but it looks good to go.

**Pull request merged**